lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -925,33 +933,40 @@ findMatchingConcatDimForSameSizeSubTensor(llvm::ArrayRef<size_t> firstDims,
 /// there is a dimension in the input tensors, with which we can meet
 /// two requirements:
 /// 1) Input tensors are concatenate-able along this dimension.
-/// 2) The sub-tensors after this dimension in the input tensors, are
-///    of the same size as the sub-tensor of the input of the original
-///    Concat node after the concatenation dimension.
+/// 2) The trailing/leading dimensions sizes after/before this dimension in the
+///    input tensors, are of the same size as the trailing/leading dimensions of
+///    the input of the original Concat node after/before the concatenation
+///    dimension.
```

Comment:
Can you explain why we need to meet these conditions? E.g. because they ensure that the payload of the new concat node should be the same as the payload of the original concat node.